app/models/spree/variant_decorator.rb

```diff
@@ -18,11 +18,13 @@
   attr_accessible :unit_value, :unit_description, :images_attributes, :display_as, :display_name, :import_date
   accepts_nested_attributes_for :images
 
-  validates_presence_of :unit_value,
-    if: -> v { %w(weight volume).include? v.product.andand.variant_unit }
+  validates :unit_value, presence: true, if: -> (variant) {
+    %w(weight volume).include?(variant.product.andand.variant_unit)
+  }
```

Comment:
This is the validation copied in `validate_presence_of_unit_value`. See below why this does not work for `ProductSet`:

```ruby
   41:       product.assign_attributes(attributes.except(*variant_related_attrs))
   42:
   43:       byebug
=> 44:       product.variants.each do |variant|
   45:         validate_presence_of_unit_value(product, variant)
   46:       end
   47:

...
(byebug) attributes
{:id=>618, :variant_unit=>"weight", :variant_unit_scale=>1}
(byebug) product.variant_unit
"weight"
(byebug) product.variants.first.product.variant_unit
"items"
```

We did update the value for the product object but this didn't affect the variant's value as they are different Ruby objects.

```
(byebug) product.object_id == product.variants.first.product.object_id
false
```